trino/trino_test.go

```diff
@@ -765,6 +765,106 @@ func TestQueryColumns(t *testing.T) {
 	assert.Equal(t, actualTypes, expectedTypes)
 }
 
+func TestParametricDateTime(t *testing.T) {
+	c := &Config{
+		ServerURI:         *integrationServerFlag,
+		SessionProperties: map[string]string{"query_priority": "1"},
+	}
+
+	dsn, err := c.FormatDSN()
+	require.NoError(t, err)
+
+	db, err := sql.Open("trino", dsn)
+	require.NoError(t, err)
+
+	t.Cleanup(func() {
+		assert.NoError(t, db.Close())
+	})
+
+	rows, err := db.Query(`SELECT
+  cast(current_time as time(6)) AS timep,
+  cast(current_time as time(6) with time zone) AS timeptz,
+  cast(current_time as timestamp(6)) AS tsp,
+  cast(current_time as timestamp(6) with time zone) AS tsptz`,
```

Comment:
What's the expected behavior for a high precision, like 12? If we can't store such values and they would be truncated, an error should be raised, or they should be returned as strings.

Reply:
We now error if the precision is greater than nanoseconds